src/sync-controller.js

```diff
@@ -46,35 +46,33 @@ export const syncPointStrategies = [
         const datetimeMapping =
           syncController.timelineToDatetimeMappings[segment.timeline];
 
-        if (!datetimeMapping) {
+        if (!datetimeMapping || !segment.dateTimeObject) {
           continue;
         }
 
-        if (segment.dateTimeObject) {
-          const segmentTime = segment.dateTimeObject.getTime() / 1000;
-          let start = segmentTime + datetimeMapping;
+        const segmentTime = segment.dateTimeObject.getTime() / 1000;
+        let start = segmentTime + datetimeMapping;
 
-          // take part duration into account.
-          if (segment.parts && typeof partAndSegment.partIndex === 'number') {
-            for (let z = 0; z < partAndSegment.partIndex; z++) {
-              start += segment.parts[z].duration;
-            }
-          }
-          const distance = Math.abs(currentTime - start);
-
-          // Once the distance begins to increase, or if distance is 0, we have passed
-          // currentTime and can stop looking for better candidates
-          if (lastDistance !== null && (distance === 0 || lastDistance < distance)) {
-            break;
+        // take part duration into account.
+        if (segment.parts && typeof partAndSegment.partIndex === 'number') {
+          for (let z = 0; z < partAndSegment.partIndex - 1; z++) {
```

Comment:
This function is the same except for the short circuit that I added and this line which only considers durations before the current part index, because we are trying to find what part **starts** at a time.